### PR TITLE
`Development`: Speed up client tests for programming exercise assessment manual result

### PR DESCRIPTION
**Pull request by MerlinMehmed**
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x]  I ran the tests locally and all tests passed.

### Motivation and Context

Speeding up the client programming-assessment-manual-result component test.
Unfortunately, I couldn't solve the memory leaks. If you know how to do it, feel free to commit to this PR.

#### Code Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts [hunk 1]
```diff
@@ -2,22 +2,16 @@ import * as ace from 'brace';
...
 import { MockParticipationWebsocketService } from '../../helpers/mocks/service/mock-participation-websocket.service';
-import { BuildLogService } from 'app/exercises/programming/shared/service/build-log.service';
 import { User } from 'app/core/user/user.model';
 import { AccountService } from 'app/core/auth/account.service';
 import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
 import { By } from '@angular/platform-browser';
-import { AlertService } from 'app/core/util/alert.service';
-import { MockComponent, MockPipe } from 'ng-mocks';
-import { ResultService } from 'app/exercises/shared/result/result.service';
+import { MockComponent, MockPipe, MockProvider, MockDirective } from 'ng-mocks';
```
Comment (Olimbia23) [inline src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:14]: client-style issue

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts [hunk 2]
```diff
@@ -229,81 +217,80 @@ describe('CodeEditorTutorAssessmentContainerComponent', () => {
...
-        expect(comp.complaint).to.not.exist;
+        expect(getIdentityStub).toHaveBeenCalledTimes(1);
+        expect(lockAndGetProgrammingSubmissionParticipationStub).toHaveBeenCalledTimes(1);
+        expect(findBySubmissionIdStub).toHaveBeenCalledTimes(1);
+        expect(comp.complaint).toBe(undefined);
         fixture.detectChanges();
 
         const complaintsForm = debugElement.query(By.css('jhi-complaints-for-tutor-form'));
-        expect(complaintsForm).to.not.exist;
+        expect(complaintsForm).toBe(null);
```
Comment (Olimbia23) [inline src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:289]: [nit] `toBe(null)` or `toBe(undefined)`
I am unsure which one is our convention, but I guess it makes sense to stick with one approach.

**Review by Olimbia23 — DISMISSED** [review]
Nice work @MerlinMehmed
Code changes look good.
Approve 👍 

Ps. left two minor comments.

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts (hunk 2 above)
Comment (imstreetspirit) [reply src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:289]: I had same the case in my PR and I learned that null is correct one: https://github.com/ls1intum/Artemis/pull/4104#discussion_r743950553

Comment (krusche) [reply src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:289]: we always prefer `undefined` over `null` in the client

**Review by coolchock — APPROVED** [review]
looks good :+1:

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts (hunk 2 above)
Comment (MerlinMehmed) [reply src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:289]: The [Sinon/Chai to Jest Conversion](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=105252048) and the [Client Tests](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/) define the following: 
Screenshot from 2021-11-07 09-36-07
Screenshot from 2021-11-07 09-38-58
Shall we change that? In the current case if I change `null` to `undefined` the test fails.

**Comment by MerlinMehmed** [comment]
Thank you @Olimbia23 and @coolchock for the review. I applied Olimbia's comments and it would be great if you can reapprove.

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts (hunk 2 above)
Comment (krusche) [reply src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts:289]: if it is a CSS value, you can use toBe(null).

**Review by imstreetspirit — APPROVED** [review]
it looks good to me

**Review by Olimbia23 — APPROVED** [review]
Thank you for applying the suggestions @MerlinMehmed 
Code changes look good.
Approve 👍

**Pull request merged**